backend/src/api.py

```diff
@@ -174,11 +178,34 @@ def toDict(self):
         }
 
 
+@dataclass
+class Dependency:
+    display_name: str
+    pypi_name: str
+    version: str
+    size_estimate: int | float
+    auto_update: bool = False
```

Comment:
`auto_update` is only used for NCNN. Why?

Reply:
It's like this currently in chaiNNer. since we control the NCNN package and have introduced features side-by-side with chaiNNer updates, we figured having it auto update (if installed) was the right call to avoid issues. We obviously don't want this happening with something like pytorch and it's not really necessary for anything else. But since we did things like change our ncnn implementation to use new functions we added to the bindings, we need ncnn to stay up to date

Reply:
Is the version pinning we do not good enough for that?

Reply:
It's an optional dependency, so a user has to manually initiate an update for it to install it at all.

Reply:
but with this auto update flag, if the dependency is installed, it will then from that point on auto update if there is an update to perform, without the user needing to manually trigger it. this is to prevent ncnn from becoming outdated